### PR TITLE
Added support for http user basic authentication

### PR DESCRIPTION
**Pull request by moisespsena**
Added support for http user basic authentication

### CODE REVIEW

File: _examples/custom_http_auth/main.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		Timeout: 15 * time.Second,
+
+		// don't follow redirect
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	// Override http(s) default protocol to use our custom client
+	client.InstallProtocol("https", githttp.NewClient(customClient))
```
Comment (orirawlings) [inline _examples/custom_http_auth/main.go]: Apologies in advance if this is merely my own ignorance, but is there a specific reason we need to include this in the example code? It seems to me like encouraging people to disable SSL verification and redirects isn't good general practice. I think this is more or less copy/pasted from the existing `custom_http` example but maybe doesn't make sense here.

For this example code I think it makes more sense to just focus on demonstrating the basic http auth usage in isolation.

File: _examples/custom_http_auth/main.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package main
```
Comment (orirawlings) [inline _examples/custom_http_auth/main.go]: I'm thinking we might want to rename this example from `custom_http_auth` to `http_basic_auth`

Comment (orirawlings) [inline _examples/custom_http_auth/main.go]: We should add a link to the example in the `_examples/README.md`

File: _examples/custom_http_auth/main.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	"gopkg.in/src-d/go-git.v4"
+	. "gopkg.in/src-d/go-git.v4/_examples"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/client"
+	githttp "gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+// Here is an example to configure http client according to our own needs.
+func main() {
+	CheckArgs("<url>", "<username>", "<password>")
```
Comment (orirawlings) [inline _examples/custom_http_auth/main.go]: To get the tests for the example code to pass, you'll need to set some arguments in the test file: https://github.com/src-d/go-git/blob/master/_examples/common_test.go#L17

```
=== RUN   TestExamples/custom_http_auth
Usage: /tmp/go-build169800819/command-line-arguments/_obj/exe/main <url> <username> <password>
exit status 1
```

File: _examples/custom_http_auth/main.go (hunk 2 above)
Comment (moisespsena) [inline _examples/custom_http_auth/main.go]: It's a good idea.

Comment (moisespsena) [inline _examples/custom_http_auth/main.go]: Excelent!

File: _examples/custom_http_auth/main.go (hunk 3 above)
Comment (moisespsena) [inline _examples/custom_http_auth/main.go]: Ok.

File: _examples/custom_http_auth/main.go (hunk 1 above)
Comment (moisespsena) [inline _examples/custom_http_auth/main.go]: I agree.

File: plumbing/transport/http/common.go [hunk 4]
```diff
@@ -145,6 +145,27 @@ type AuthMethod interface {
...
+func (a *BasicAuthMethod) String() string {
+	return a.auth.String()
+}
+
+func (a *BasicAuthMethod) setAuth(r *http.Request)  {
+	a.auth.setAuth(r)
+}
+
+func NewBasicAuthMethod(username, password string) *BasicAuthMethod {
+	return &BasicAuthMethod{auth:NewBasicAuth( "moisespsena", "ghis5509")}
```
Comment (mcuadros) [inline plumbing/transport/http/common.go]: err, this is a user and a password...

**Comment by mcuadros** [comment]
Can you explain the goals of this PR, why is not enough [`http.BasicAuth`](https://godoc.org/gopkg.in/src-d/go-git.v4/plumbing/transport/http#BasicAuth)

File: plumbing/transport/http/common.go (hunk 4 above)
Comment (moisespsena) [inline plumbing/transport/http/common.go]: Thanks.

**Comment by mcuadros** [comment]
close due to inactivity

**Pull request closed without merging**